### PR TITLE
[Flaky Test] Fix test_gluon_model_zoo.test_models when MXNET_MKLDNN_DEBUG=1 

### PR DESCRIPTION
**Pull request by azai91**
## Description ##
Fix MKLDNN debug issue in PR https://github.com/apache/incubator-mxnet/issues/10026

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Use functional flatten method as opposed to built in
- [x] Increase tolerance for similar ndarray
- [x] Use reordered inputs
- [x] Re-enable MXNET_MKLDNN_DEBUG flag in CI
- [x] Added exclude flag for LRN backwards operator as fcompute vs fcomputex operators accepts different number of inputs

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by pengzhao-intel** [comment]
Thanks, @azai91, LGTM.

BTW, please add the env in the CI and let's see if all CI can pass.
https://github.com/apache/incubator-mxnet/pull/10069

**Comment by azai91** [comment]
@pengzhao-intel I had to exclude LRN operator from being tested as the fcompute and fcomputeex functions accept different number of inputs (fcompute requires norm from forward while fcomputeex does not). added a flag to the attribute to exclude debugging just this operator.

**Comment by pengzhao-intel** [comment]
Yes, it's different. Agree to exclude it for debugging check.

File: src/operator/nn/mkldnn/mkldnn_base.cc [hunk 1]
```diff
@@ -496,11 +496,8 @@ void OpCheck::Run(mxnet::FCompute fn, const nnvm::NodeAttrs &attrs,
                   const std::vector<mxnet::NDArray> &inputs_,
                   const std::vector<mxnet::OpReqType> &req,
                   const std::vector<mxnet::NDArray> &outputs_) {
-
   static auto& is_excluded = Op::GetAttr<bool>("TExcludeMKLDNNDebug");
-
   if (is_excluded.get(attrs.op, false)) return;
-
```
Comment (pengzhao-intel) [inline src/operator/nn/mkldnn/mkldnn_base.cc]: Could we give a warning that the OP is not checked?

**Comment by azai91** [comment]
@pengzhao-intel review/approve please

**Comment by pengzhao-intel** [comment]
LGTM, thanks for the fix :)

**Comment by azai91** [comment]
@anirudh2290 stamp plz

**Pull request merged**